packages/components/src/popover/index.js

```diff
@@ -42,405 +46,71 @@ const SLOT_NAME = 'Popover';
 
 const slotNameContext = createContext();
 
-function computeAnchorRect(
-	anchorRefFallback,
-	anchorRect,
-	getAnchorRect,
-	anchorRef = false,
-	container
-) {
-	if ( anchorRect ) {
-		return anchorRect;
-	}
-
-	if ( getAnchorRect ) {
-		if ( ! anchorRefFallback.current ) {
-			return;
-		}
-
-		const rect = getAnchorRect( anchorRefFallback.current );
-		return offsetIframe(
-			rect,
-			rect.ownerDocument || anchorRefFallback.current.ownerDocument,
-			container
-		);
-	}
-
-	if ( anchorRef !== false ) {
-		if (
-			! anchorRef ||
-			! window.Range ||
-			! window.Element ||
-			! window.DOMRect
-		) {
-			return;
-		}
-
-		// Duck-type to check if `anchorRef` is an instance of Range
-		// `anchorRef instanceof window.Range` checks will break across document boundaries
-		// such as in an iframe.
-		if ( typeof anchorRef?.cloneRange === 'function' ) {
-			return offsetIframe(
-				getRectangleFromRange( anchorRef ),
-				anchorRef.endContainer.ownerDocument,
-				container
-			);
-		}
+const positionToPlacement = ( position ) => {
+	const [ x, y, z ] = position.split( ' ' );
 
-		// Duck-type to check if `anchorRef` is an instance of Element
-		// `anchorRef instanceof window.Element` checks will break across document boundaries
-		// such as in an iframe.
-		if ( typeof anchorRef?.getBoundingClientRect === 'function' ) {
-			const rect = offsetIframe(
-				anchorRef.getBoundingClientRect(),
-				anchorRef.ownerDocument,
-				container
-			);
-
-			return rect;
+	if ( [ 'top', 'bottom' ].includes( x ) ) {
+		let suffix = '';
+		if ( ( !! z && z === 'left' ) || y === 'right' ) {
+			suffix = '-start';
+		} else if ( ( !! z && z === 'right' ) || y === 'left' ) {
+			suffix = '-end';
 		}
-
-		const { top, bottom } = anchorRef;
-		const topRect = top.getBoundingClientRect();
-		const bottomRect = bottom.getBoundingClientRect();
-
-		return offsetIframe(
```

Comment:
I remember that there were issues calculating the position in iframes and right now the block toolbar for example in site editor is not well positioned.

https://user-images.githubusercontent.com/16275880/166719968-315f4575-1da5-40e4-9ab1-e8f80a6e4d9b.mov

Reply:
solved.